napari/_vispy/vendored/volume.py

```diff
@@ -1123,7 +1142,8 @@ def attenuation(self):
     @attenuation.setter
     def attenuation(self, value):
         self._attenuation = float(value)
-        self.shared_program['u_attenuation'] = self._attenuation
+        if 'u_attenuation' in self.shared_program:
+            self.shared_program['u_attenuation'] = self._attenuation
         self.update()
```

Comment:
hmm yeah this seems a little fishy, maybe you can look at PR where this was added if there was any discussion here. I can try taking a look too

Reply:
#1454 is the first, it seems. At that point, there were two assert rather than one. I never used attenuation, so I'm not sure what the expected behaviour should be here. Does it depend on the defaults?

Reply:
I reverted this change here for now, and opened an issue specifically for it at #3301 .